File: go/raindrops/raindrops.go
package raindrops

import "strconv"

const testVersion = 2

func Convert(n int) (ret string) {
	if n%3 == 0 {
		ret += "Pling"
	}
	if n%5 == 0 {
		ret += "Plang"
	}
	if n%7 == 0 {
		ret += "Plong"
	}
	if len(ret) == 0 {
		ret = strconv.Itoa(n)
	}
	return
}

File: scala/raindrops/src/main/scala/Raindrops.scala
import scala.collection.SortedMap

object Raindrops {
  private val sound = SortedMap(
    3 -> "Pling",
    5 -> "Plang",
    7 -> "Plong"
  )

  def convert(n: Int): String = {
    val drops = sound.filterKeys(n % _ == 0).values.mkString
    if (drops.isEmpty) n.toString
    else drops
  }

}

